features/events/syslog/src/main/java/org/opennms/netmgt/syslogd/SyslogSinkConsumer.java

```diff
@@ -0,0 +1,210 @@
+                }
+            }
+        }
+    }
+
+    public Log toEventLog(SyslogMessageLogDTO messageLog) {
+        final Log elog = new Log();
+        final Events events = new Events();
+        elog.setEvents(events);
+        for (SyslogMessageDTO message : messageLog.getMessages()) {
+            try {
+                LOG.debug("Converting syslog message into event.");
+                ConvertToEvent re = new ConvertToEvent(
+                        messageLog.getSystemId(),
+                        messageLog.getLocation(),
+                        messageLog.getSourceAddress(),
+                        messageLog.getSourcePort(),
+                        // Decode the packet content as ASCII
+                        // TODO: Support more character encodings?
+                        StandardCharsets.US_ASCII.decode(message.getBytes()).toString(),
+                        syslogdConfig
+                    );
+                events.addEvent(re.getEvent());
+            } catch (final UnsupportedEncodingException e) {
+                LOG.info("Failure to convert package", e);
+            } catch (final MessageDiscardedException e) {
+                LOG.info("Message discarded, returning without enqueueing event.", e);
+            } catch (final Throwable e) {
+                LOG.error("Unexpected exception while processing SyslogConnection", e);
+            }
+        }
+        return elog;
+    }
+
+    private void broadcast(Log eventLog)  {
+        if (LOG.isTraceEnabled())  {
+            for (Event event : eventLog.getEvents().getEventCollection()) {
+                LOG.trace("Processing a syslog to event dispatch", event.toString());
+                String uuid = event.getUuid();
+                LOG.trace("Event {");
+                LOG.trace("  uuid  = {}", (uuid != null && uuid.length() > 0 ? uuid : "<not-set>"));
+                LOG.trace("  uei   = {}", event.getUei());
+                LOG.trace("  src   = {}", event.getSource());
+                LOG.trace("  iface = {}", event.getInterface());
+                LOG.trace("  time  = {}", event.getTime());
+                LOG.trace("  Msg   = {}", event.getLogmsg().getContent());
+                LOG.trace("  Dst   = {}", event.getLogmsg().getDest());
+                List<Parm> parms = (event.getParmCollection() == null ? null : event.getParmCollection());
+                if (parms != null) {
+                    LOG.trace("  parms {");
+                    for (Parm parm : parms) {
+                        if ((parm.getParmName() != null)
+                                && (parm.getValue().getContent() != null)) {
+                            LOG.trace("    ({}, {})", parm.getParmName().trim(), parm.getValue().getContent().trim());
+                        }
+                    }
+                    LOG.trace("  }");
+                }
+                LOG.trace("}");
+            }
+        }
+        eventForwarder.sendNowSync(eventLog);
```

Comment:
Unlike other event producers, the messages handled by this consumer can live in an external broker.
We send synchronously in order to perform as much processing as possible on the current thread before grabbing another message.